Foundry Unable to Start - Errors During Unzip

Hello,

I'm currently having trouble getting the container to start up since pulling the latest image. Previously, I was on a version of the image from 2022-01-26 (image does not have a tag) without issue, but since updating the image and the yml file to the new specs, I've been unable to get the container to start. I've even tried running a new version of the container. The below details are all from the new container.

I originally configured this container using the guidance of https://github.com/marcwagner/foundry_swag_docker but have since modified the docker-compose.yml to match the changes made to the image about 10 days ago.

Relevant portion of the docker-compose.yml (really the stack config in portainer):
```yml
version: "3.8"
services:
  foundry:
    container_name: foundryCurrent
    hostname: foundry
    image: direckthit/fvtt-docker:latest
    expose:
      - 30000
    volumes:
      - /dockerVolumes/foundryCurrent/data:/data/foundryvtt     # replace with the correct path to your foundry config dir
      - /dockerVolumes/foundryCurrent/downloads:/host                     # replace with the correct path to your foundry-x.x.x.zip file
    restart: unless-stopped
```

Log snippet - this repeats continuously. The container is unable to extract to /opt/foundryvtt/resources/app/common/ but from what I've been able to decipher from run-server.sh, it should be able to
```
2022-08-19T03:04:39.141149844Z Archive:  foundryvtt-9.266.zip
2022-08-19T03:04:42.093754836Z   inflating: LICENSE.electron.txt
2022-08-19T03:04:42.093776611Z   inflating: LICENSES.chromium.html
2022-08-19T03:04:42.093783322Z   inflating: chrome-sandbox
2022-08-19T03:04:42.093788556Z   inflating: chrome_100_percent.pak
2022-08-19T03:04:42.093793366Z   inflating: chrome_200_percent.pak
2022-08-19T03:04:42.093798152Z   inflating: chrome_crashpad_handler
2022-08-19T03:04:42.093802921Z   inflating: foundryvtt
2022-08-19T03:04:42.093807426Z   inflating: icudtl.dat
2022-08-19T03:04:42.093811900Z   inflating: libEGL.so
2022-08-19T03:04:42.093816589Z   inflating: libGLESv2.so
2022-08-19T03:04:42.093821572Z   inflating: libffmpeg.so
2022-08-19T03:04:42.093826333Z   inflating: libvk_swiftshader.so
2022-08-19T03:04:42.093831036Z   inflating: libvulkan.so.1
2022-08-19T03:04:42.093835717Z   inflating: locales/am.pak
2022-08-19T03:04:42.093840435Z   inflating: locales/ar.pak
2022-08-19T03:04:42.093845173Z   inflating: locales/bg.pak
2022-08-19T03:04:42.093849907Z   inflating: locales/bn.pak
2022-08-19T03:04:42.093854705Z   inflating: locales/ca.pak
2022-08-19T03:04:42.093859652Z   inflating: locales/cs.pak
2022-08-19T03:04:42.093864534Z   inflating: locales/da.pak
2022-08-19T03:04:42.093869324Z   inflating: locales/de.pak
2022-08-19T03:04:42.093874595Z   inflating: locales/el.pak
2022-08-19T03:04:42.093879335Z   inflating: locales/en-GB.pak
2022-08-19T03:04:42.093884072Z   inflating: locales/en-US.pak
2022-08-19T03:04:42.093888848Z   inflating: locales/es-419.pak
2022-08-19T03:04:42.093893592Z   inflating: locales/es.pak
2022-08-19T03:04:42.093898432Z   inflating: locales/et.pak
2022-08-19T03:04:42.093903159Z   inflating: locales/fa.pak
2022-08-19T03:04:42.093907691Z   inflating: locales/fi.pak
2022-08-19T03:04:42.093921738Z   inflating: locales/fil.pak
2022-08-19T03:04:42.093927141Z   inflating: locales/fr.pak
2022-08-19T03:04:42.093931909Z   inflating: locales/gu.pak
2022-08-19T03:04:42.093936547Z   inflating: locales/he.pak
2022-08-19T03:04:42.093941172Z   inflating: locales/hi.pak
2022-08-19T03:04:42.093946124Z   inflating: locales/hr.pak
2022-08-19T03:04:42.217049921Z unzip: can't create directory 'resources/app/common/': Permission denied
2022-08-19T03:04:42.217050065Z   inflating: locales/hu.pak
2022-08-19T03:04:42.217085832Z   inflating: locales/id.pak
2022-08-19T03:04:42.217089479Z   inflating: locales/it.pak
2022-08-19T03:04:42.217092980Z   inflating: locales/ja.pak
2022-08-19T03:04:42.217096444Z   inflating: locales/kn.pak
2022-08-19T03:04:42.217099967Z   inflating: locales/ko.pak
2022-08-19T03:04:42.217103279Z   inflating: locales/lt.pak
2022-08-19T03:04:42.217106608Z   inflating: locales/lv.pak
2022-08-19T03:04:42.217110003Z   inflating: locales/ml.pak
2022-08-19T03:04:42.217113314Z   inflating: locales/mr.pak
2022-08-19T03:04:42.217116587Z   inflating: locales/ms.pak
2022-08-19T03:04:42.217119849Z   inflating: locales/nb.pak
2022-08-19T03:04:42.217123100Z   inflating: locales/nl.pak
2022-08-19T03:04:42.217126522Z   inflating: locales/pl.pak
2022-08-19T03:04:42.217129795Z   inflating: locales/pt-BR.pak
2022-08-19T03:04:42.217133061Z   inflating: locales/pt-PT.pak
2022-08-19T03:04:42.217136526Z   inflating: locales/ro.pak
2022-08-19T03:04:42.217139905Z   inflating: locales/ru.pak
2022-08-19T03:04:42.217143148Z   inflating: locales/sk.pak
2022-08-19T03:04:42.217146406Z   inflating: locales/sl.pak
2022-08-19T03:04:42.217149644Z   inflating: locales/sr.pak
2022-08-19T03:04:42.217153008Z   inflating: locales/sv.pak
2022-08-19T03:04:42.217156311Z   inflating: locales/sw.pak
2022-08-19T03:04:42.217159542Z   inflating: locales/ta.pak
2022-08-19T03:04:42.217162784Z   inflating: locales/te.pak
2022-08-19T03:04:42.217166005Z   inflating: locales/th.pak
2022-08-19T03:04:42.217169236Z   inflating: locales/tr.pak
2022-08-19T03:04:42.217172460Z   inflating: locales/uk.pak
2022-08-19T03:04:42.217175709Z   inflating: locales/vi.pak
2022-08-19T03:04:42.217179059Z   inflating: locales/zh-CN.pak
2022-08-19T03:04:42.217182320Z   inflating: locales/zh-TW.pak
2022-08-19T03:04:42.217185561Z   inflating: resources.pak
2022-08-19T03:04:42.217188844Z    creating: resources/app/common/
2022-08-19T03:04:42.262281839Z node:internal/modules/cjs/loader:936
2022-08-19T03:04:42.262322595Z   throw err;
2022-08-19T03:04:42.262327382Z   ^
2022-08-19T03:04:42.262330790Z 
2022-08-19T03:04:42.262334425Z Error: Cannot find module '/opt/foundryvtt/resources/app/main.js'
2022-08-19T03:04:42.262338045Z     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
2022-08-19T03:04:42.262341560Z     at Function.Module._load (node:internal/modules/cjs/loader:778:27)
2022-08-19T03:04:42.262345053Z     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)
2022-08-19T03:04:42.262348604Z     at node:internal/main/run_main_module:17:47 {
2022-08-19T03:04:42.262352057Z   code: 'MODULE_NOT_FOUND',
2022-08-19T03:04:42.262355400Z   requireStack: []
2022-08-19T03:04:42.262358789Z }
2022-08-19T03:05:42.832445995Z Archive:  foundryvtt-9.266.zip
```

Any help that you're able to provide is appreciated.

Yeah, I'm not  sure what is going on. I'm not well versed in docker. Getting these errors from the Portainer fvtt container log:  


> inflating: resources.pak
>    creating: resources/app/client/
> unzip: can't create directory 'resources/app/client/': Permission denied
> node:internal/modules/cjs/loader:936   
>   throw err;
>   ^
> Error: Cannot find module '/opt/foundryvtt/resources/app/main.js'
>     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
>     at Function.Module._load (node:internal/modules/cjs/loader:778:27)
>     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)
>     at node:internal/main/run_main_module:17:47 {
>   code: 'MODULE_NOT_FOUND',
>   requireStack: []

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,7 +1,7 @@
 FROM node:16-alpine
 
 RUN deluser node && \
-    mkdir /opt/foundryvtt && \
+    mkdir -p /opt/foundryvtt/resources/app && \
     mkdir /data && \
     mkdir /data/foundryvtt && \
     adduser --disabled-password fvtt && \
@@ -13,6 +13,15 @@ USER fvtt
 
 COPY --chown=fvtt run-server.sh /opt/foundryvtt
 RUN chmod +x /opt/foundryvtt/run-server.sh
+
+# Set permissions for the volumes
+USER root
+RUN chown -R fvtt:fvtt /data/foundryvtt && \
+    chown -R fvtt:fvtt /opt/foundryvtt/resources/app && \
+    chmod -R g+s /data/foundryvtt && \
+    chmod -R g+s /opt/foundryvtt/resources/app
+
+USER fvtt
 VOLUME /data/foundryvtt
 VOLUME /host
 VOLUME /opt/foundryvtt/resources/app

--- a/run-server.sh
+++ b/run-server.sh
@@ -3,7 +3,11 @@
 # look for a file name foundryvtt-.....zip or FoundryVTT...zip, copy it to 
 # /opt/foundryvtt, unzip and remove it. Then start the foundry app
 
-cd /opt/foundryvtt
-find /host -type f -name [f,F]oundry[vtt,VTT]*.zip -exec cp '{}' . ';'
-unzip -o *.zip && rm *.zip
+cd /opt/foundryvtt || exit
+echo "Copying zip file..."
+find /host -type f -name '[f,F]oundry[vtt,VTT]*.zip' -exec cp '{}' . ';'
+
+echo "Unzipping..."
+unzip -o ./*.zip && rm ./*.zip
+
 node resources/app/main.js --dataPath=/data/foundryvtt

